examples/filter.py

```diff
@@ -0,0 +1,48 @@
+'''
+Takes in any data from a sensor and filters it.
+'''
+
+import json
+import zmq
+
+
+def bayes_filter(prev, measured):
+    ''' Super lame Bayes filter'''
+    return (prev + measured)/2
+
+
+def main():
+    # {{{ wrap this stuff
+    context = zmq.Context()
+    subscriber = context.socket(zmq.SUB)
+    subscriber.connect("tcp://localhost:5000")
+    subscriber.setsockopt(zmq.SUBSCRIBE, b"sensor")
+
+    kill_sig = context.socket(zmq.SUB)
+    kill_sig.connect("tcp://localhost:9000")
+    kill_sig.setsockopt(zmq.SUBSCRIBE, b"EXIT")
+    # }}}
```

Comment:
I think this code should explain better than I can in words:

```py
socket = context.socker(zmq.SUB)
socket.connect('tcp://localhost:9000')
socket.setsockopt(zmq.SUBSCRIBE, b'EXIT')
socket.connect('tcp://localhost:4000')
socket.setsockopt(zmq.SUBSCRIBE, b'sensor')
```

If the documentation serves me correctly we can also replace calls like this: `socket.setsockopt(zmq.SUBSCRIBE, b'foo')` with `socket.subscribe(b'foo')`